core/vm.cpp

```diff
@@ -1839,16 +1849,49 @@ class Interpreter {
                             goto replaceframe;
                         }
                     }
-                    // Equality can be used when the types don't match.
                     switch (ast.op) {
+                        // Equality can be used when the types don't match.
                         case BOP_MANIFEST_EQUAL:
                         std::cerr << "INTERNAL ERROR: Equals not desugared" << std::endl;
                         abort();
 
+                        // Equality can be used when the types don't match.
                         case BOP_MANIFEST_UNEQUAL:
                         std::cerr << "INTERNAL ERROR: Notequals not desugared" << std::endl;
                         abort();
 
+                        // e in e
+                        case BOP_IN: {
+                            if (lhs.t != Value::STRING) {
+                                throw makeError(ast.location,
+                                                "The left hand side of the 'in' operator should be "
+                                                "a string,  got " + type_str(lhs));
+                            }
+                            auto *field = static_cast<HeapString*>(lhs.v.h);
+                            switch (rhs.t) {
+                                case Value::ARRAY: {
+                                    auto *arr = static_cast<HeapArray*>(rhs.v.h);
+                                    // TODO(dcunnin): Implement this.
+                                    (void) arr;
+                                    throw makeError(ast.location, "e in [...] not implemented yet.");
+                                } break;
+
+                                case Value::OBJECT: {
+                                    auto *obj = static_cast<HeapObject*>(rhs.v.h);
+                                    auto *fid = alloc->makeIdentifier(field->value);
+                                    unsigned unused_found_at = 0;
+                                    bool in = findObject(fid, obj, 0, unused_found_at);
+                                    scratch = makeBoolean(in);
+                                } break;
+  
+                                default:
+                                throw makeError(ast.location,
```

Comment:
I checked Python, it's annoying in that [1, 2, 3] is not in [0, 1, 2, 3, 4] but "123" is in "01234".  So let's punt string as well.